### DISCUSSION TITLE
Where are Entity Field List Values stored

### DISCUSSION THREAD

**Discussion by paultvis** [atrocore/atrocore#603] [Q&A] [answered]
Hi,

Can someone tell me where entity list values live in the database? We have a field we have added to products and I want to get the values directly from the db. The field values on the products have been set to a default value, but I cannot see anything in the db with that value. 

Any help appreciated.

Thanks

**Comment by rratsun** [comment, marked as answer]
I don't recommend you to get data directly from DB. It's a bad idea. Please, use REST API. 

Tables schemas can be changed because of optimizations reasons or other reasons. In PIM also can be some virtualizations: when you see more entity data than entity db table have. We are use some tricks for make PIM system quicker and more customizable, that's why we strongly recommend you to use API and don't try to get data directly from DB.

**Reply by paultvis** [reply]
Hi @rratsun, thank you for your feedback. I am well aware of the risks and challenges in pulling data directly from the db. However, the source system we are sending the data to needs translation of some of the values at a technical level. Until we are able to implement a proper ETL solution it makes more sense to pull directly from the db and manipulate. This is an interim solution and with our experience of dbs we know and accept the risks. 

With that in mind, can you tell me where these values are stored?

Thanks

**Reply by paultvis** [reply]
@rratsun, we never progressed this task but this is now one of the last remaining tasks to get our new website live. We have created a field on products for shipping code. This is a list field and we need to get the available values to set up the attribute on the website. I am still unable to find these values in the database, and I have looked at the API but that also does not appear to have an endpoint which will give me the list of available values in the field list. Can you tell me which endpoint to use to get this please if we cannot get from the database directly?

Thanks

**Reply by rzablodskyi** [reply]
Hello @paultvis. If I have correctly understood your issue and you need to get all available options of list type field in Product, you can use such API endpoint - `/Metadata`, request method `GET`. Needed options stored in `entityDefs.Product.fields.{yourFieldName}` in `options` key.
